Libplanet.Explorer/Queries/StateQuery.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Security.Cryptography;
+using GraphQL;
+using GraphQL.Types;
+using Libplanet.Action;
+using Libplanet.Assets;
+using Libplanet.Blockchain;
+using Libplanet.Blockchain.Policies;
+using Libplanet.Blocks;
+using Libplanet.Explorer.GraphTypes;
+
+namespace Libplanet.Explorer.Queries;
+
+public class StateQuery<T>
+    : ObjectGraphType<(IBlockChainStates<T> ChainStates, IBlockPolicy<T> Policy)>
+    where T : IAction, new()
+{
+    public StateQuery()
+    {
+        Name = "StateQuery";
+        Field<NonNullGraphType<FungibleAssetValueType>>(
+            "balance",
+            arguments: new QueryArguments(
+                new QueryArgument<NonNullGraphType<AddressType>> { Name = "owner" },
+                new QueryArgument<NonNullGraphType<ByteStringType>> { Name = "currencyHash" },
+                new QueryArgument<NonNullGraphType<IdGraphType>> { Name = "offsetBlockHash" }
+            ),
+            resolve: ResolveBalance
+        );
+        Field<FungibleAssetValueType>(
+            "totalSupply",
+            arguments: new QueryArguments(
+                new QueryArgument<NonNullGraphType<ByteStringType>> { Name = "currencyHash" },
+                new QueryArgument<NonNullGraphType<IdGraphType>> { Name = "offsetBlockHash" }
+            ),
+            resolve: ResolveTotalSupply
+        );
+    }
+
+    private static Currency GetNativeTokenFromHash(
+        IResolveFieldContext<(IBlockChainStates<T> ChainStates, IBlockPolicy<T> Policy)> context,
+        string name
+    )
+    {
+        byte[] currencyHashBytes = context.GetArgument<byte[]>(name);
+        HashDigest<SHA1> currencyHash;
+        try
+        {
+            currencyHash = new HashDigest<SHA1>(currencyHashBytes);
+        }
+        catch (Exception e)
+        {
+            throw new ExecutionError("currencyHash must consist of 20 bytes.\n" + e.Message, e);
+        }
+
+        IImmutableSet<Currency> nativeTokens = context.Source.Policy.NativeTokens;
+        try
+        {
+            return nativeTokens.First(token => token.Hash.Equals(currencyHash));
+        }
+        catch (InvalidOperationException e)
+        {
+            throw new ExecutionError(
+                "currencyHash must be a native token.  Native tokens are:\n\n" +
+                    string.Join("\n", nativeTokens.Select(token => token.Hash.ToString())),
```

Comment:
```suggestion
                "currencyHash must be a hash of a native token.  Native tokens are:\n\n" +
                    string.Join("\n", nativeTokens.Select(token => token.Ticker + ": " + token.Hash.ToString())),
```

It might be better to have the tickers of the tokens printed along with the hash. Also, maybe consider the case of `name` having different value other than `currencyHash`.

Reply:
This message now shows tickers too.  Amended the last commit.